core/controllers/topic_editor.py

```diff
@@ -122,7 +165,13 @@ def get(self, topic_id):
 
     @acl_decorators.can_edit_topic
     def put(self, topic_id):
-        """Updates properties of the given topic."""
+        """Updates properties of the given topic.
+        Also, each change_dict given for editing should have an additional
+        property called is_topic_change, which would be a boolean. If True, it
+        means that change is for a topic, while False would mean it is for a
+        Subtopic Page (this includes adding, removing subtopics and editing its
```

Comment:
Actually -- add_subtopic and delete_subtopic are really topic-level changes, right? In the sense that subtopics is a field within Topic?

Only edit_subtopic_page is scoped to the subtopic page only.

Maybe what you want is something like change_affects_subtopic_page?

Reply:
Yeah, forgot to change that docstring. Done.